src/pytorch_lightning/utilities/types.py

```diff
@@ -141,29 +140,32 @@ def step(self, metrics: Union[float, int, Tensor], epoch: Optional[int] = None)
         ...
 
 
-# Inferred from `torch.nn.parallel.distributed.pyi`
-# Missing attributes were added to improve typing
-@runtime_checkable
-class DistributedDataParallel(Protocol):
-    def __init__(
-        self,
-        module: torch.nn.Module,
-        device_ids: Optional[List[Union[int, torch.device]]] = None,
-        output_device: Optional[Union[int, torch.device]] = None,
-        dim: int = 0,
-        broadcast_buffers: bool = True,
-        process_group: Optional[ProcessGroup] = None,
-        bucket_cap_mb: int = 25,
-        find_unused_parameters: bool = False,
-        check_reduction: bool = False,
-        gradient_as_bucket_view: bool = False,
-        static_graph: bool = False,
-    ) -> None:
-        ...
-
-    @contextmanager
-    def no_sync(self) -> Generator:
-        ...
+if torch.distributed.is_available():
+    from torch._C._distributed_c10d import ProcessGroup
```

Comment:
Can you just add an import fallback for this instead of conditionally defining the entire class?